cosi/cosi.md

```diff
@@ -258,64 +258,82 @@ steps below successfully.
 
 # Collective Signing Protocol 
 
-This section introduces the distributed CoSi protocol with n participants. For
-simplicity, we assume there is a designated leader who is
-responsible for collecting the shares and generating the signature. This leader
-could be any of the signers and is not trusted in any way. All participants are
-communicating through a reliable channel with the leader.
+This section introduces the distributed CoSi protocol with n participants. 
+The CoSi protocol has four distinct phases, and each one of these phases uses a
+distinct packet:
++ the Announcement packet for the Announcement phase
++ the Commitment packet for the Commitment phase 
++ the Challenge packet for the Challenge phase 
++ the Response packet for the Response phase 
+
+The packet formats are described in details in the section XXX Packet Format XXX.
+
+For simplicity, we assume there is a designated leader who is responsible for
+collecting the shares and generating the signature. This leader could be any of
+the signers and is not trusted in any way. All participants are communicating
+through a reliable channel with the leader.
 
 ## Collective Signature
 
-The leader must know the statement S to be signed and the set of public keys of
-the participants N. The point A is defined as the collective key of the
-participants N. A collective signature is generated in four steps over two
+The leader must know the message M to be signed and the set of public keys of
+the participants P. The point A is defined as the collective key of the
+participants P. A collective signature is generated in four steps over two
 round trips between the leader and the rest of the participants. 
 
 ### Announcement
 
-Upon the request to generate a signature on a statement S, the leader broadcasts
-an announcement message indicating the start of a signing process. It is up to
-the implementation to decide whether to send S itself during that phase or not. 
+Upon the request to generate a signature on a message M, the leader broadcasts
+an Announcement packet indicating the start of a signing process. This
+Announcement packet MUST contain the message M to sign.
```

Comment:
Another way to put it: any leader that's sending messages down for cosigning that the leader doesn't know that honest cosigners will sign is a broken leader.